core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -733,6 +745,13 @@ public FileInfo getFileInfo(AlluxioURI path, GetStatusOptions options)
         auditContext.setAllowed(false);
         throw e;
       }
+      // Possible ufs sync.
+      if (syncMetadata(journalContext, inodePath, lockingScheme)) {
+        // If synced, do not load metadata.
+        options.setLoadMetadataType(LoadMetadataType.Never);
+      }
+      inodePath.validatePath(lockingScheme);
```

Comment:
is this line always a no-op since we set `fullPathRequired=false` above? it might be cleaner to take `fullPathRequired` out of `LockingScheme` since it isn't really related to locking.

Reply:
Done. Now, the full path is only checked when required by the operation.